cached_property.py

```diff
@@ -16,23 +17,44 @@ class cached_property(object):
         Source: https://github.com/bottlepy/bottle/commit/fa7733e075da0d790d809aa3d2f53071897e6f76
         """
 
-    def __init__(self, func):
-        self.__doc__ = getattr(func, '__doc__')
+    def __init__(self, ttl=None):
+        self.ttl = ttl
+        
+    def __call__(func, doc=None):
+        self.__doc__ = doc or getattr(func, '__doc__')
+        self.__name__ = func.__name__
+        self.__module__ = func.__module__
+        
         self.func = func
+        
+        return self
 
     def __get__(self, obj, cls):
         if obj is None:
             return self
-        value = obj.__dict__[self.func.__name__] = self.func(obj)
-        return value
+        if not self.ttl:
+            value = obj.__dict__[self.func.__name__] = self.func(obj)
+            return value
+        else:
+            now = time.time()
+            key = '_cached_' + self.func.__name__
+            try:
+                value, last_update = obj.__dict__[key]
```

Comment:
Instead of using `__dict__`, you may have to use another dunder field (like `_cache`).